app/Models/Requestable.php

```diff
@@ -31,15 +29,20 @@ public function scopeRequestedBy($query, User $user)
         });
     }
 
-    public function request()
+    public function request($qty = 1)
     {
         $this->requests()->save(
-            new CheckoutRequest(['user_id' => Auth::id()])
+            new CheckoutRequest(['user_id' => Auth::id(), 'qty' => $qty])
         );
     }
 
-    public function cancelRequest()
+    public function deleteRequest()
     {
         $this->requests()->where('user_id', Auth::id())->delete();
     }
+
+    public function cancelRequest()
+    {
+        $this->requests()->where('user_id', Auth::id())->update(['canceled_at' => \Carbon\Carbon::now()]);
+    }
```

Comment:
This looks really solid, and I love that it's a `trait` - but if so, should it live in `app/traits` instead? It's not a model in and of itself, unless `use`'ed.

Reply:
I would think so, but I didn't write this file so not sure what it would break if it got moved.

Reply:
(Also Loggable, etc are in the main models directory even though they are traits. It's worth fixing, but it's unrelated to this PR I think.)